src/main/editDelete/punctuate.ts

```diff
@@ -0,0 +1,64 @@
+import { MapCallback } from '../util';
+import { Word } from '../../sharedTypes';
+
+type PartialWord = Pick<Word, 'word' | 'startTime' | 'duration'>;
+type Thresholds = { lowerCommaThreshold: number; upperCommaThreshold: number };
+
+// // TODO: use silence averaging to implement punctuation
+// const lowerCommaThreshold = 0.1;
+// const upperCommaThreshold = 0.17;
+
+const capitalizeFirstLetter: (string: string) => string = (string) => {
+  return string.charAt(0).toUpperCase() + string.slice(1);
+};
+
+const calculateThresholds: (averageSilenceDuration: number) => Thresholds = (
+  averageSilenceDuration
+) => {
+  return {
+    lowerCommaThreshold: averageSilenceDuration * 0.85, // determined constants through trial and error
+    upperCommaThreshold: averageSilenceDuration * 1.8,
+  };
+};
+
+const punctuate: (
+  totalDuration: number,
+  averageSilenceDuration: number
+) => MapCallback<PartialWord, PartialWord> =
+  (totalDuration, averageSilenceDuration) => (word, index, words) => {
+    const isLastWord = index === words.length - 1;
+    const endTime = isLastWord ? totalDuration : words[index + 1].startTime;
+    const silenceDuration = endTime - word.startTime - word.duration;
+
+    if (index === 0) {
+      words[index].word = capitalizeFirstLetter(words[index].word);
+    }
+
+    const thresholds: Thresholds = calculateThresholds(averageSilenceDuration);
+
+    let punctuation: string;
+    if (silenceDuration < thresholds.lowerCommaThreshold) {
```

Comment:
I wanted to use a nested turnery here but ESLint said no

Reply:
Good bot